docs/api/presenter.md

```diff
@@ -156,23 +156,89 @@ function StepperForm ({ count }) {
 
 ### `setup(repo, props, state)`
 
-Called when a presenter is created, useful any prep work. `setup` runs before the first `getModel` invocation.
+Called when a presenter is created, useful any prep work. `setup` runs
+before the first `getModel` invocation.
+
+```javascript
+import { getPlanets } from '../actions/planets'
+
+class PlanetsList extends Presenter {
+  setup (repo, props, state) {
+    // Important: this.model is not defined yet!
+    repo.push(getPlanets)
+  }
+  // ...
+}
+```
 
 ### `ready(repo, props, state)`
 
-Called after the presenter has run `setup` and executed the first `getModel`. This hook is useful for fetching initial data and other start tasks that need access to the model data.
+Called after the presenter has run `setup` and executed the first
+`getModel`. This hook is useful for fetching initial data and other
+start tasks that need access to the model data.
+
+```javascript
+import { getPlanets } from '../actions/planets'
+
+class PlanetsList extends Presenter {
+  getModel () {
+    return {
+      planets: state => state.planets
+    }
+  }
+  ready (repo, props, state) {
+    if (this.model.planets.length <=0) {
+      repo.push(getPlanets)
+    }
+  }
+  // ...
+}
+```
 
 ### `update(repo, props, state)`
 
 Called when a presenter gets new props. This is useful for secondary
 data fetching and other work that must happen when a Presenter receives
 new information.
 
+```javascript
+import { getPlanet } from '../actions/planets'
+
+class Planet extends Presenter {
+  getModel (props) {
+    const { planetId } = props
+
+    return {
+      planet: state => state.planets.find(planet => planet.id === planetId)
+    }
+  }
+  // These are the next props and state
+  update (repo, props, state) {
+    if (props.planetId !== this.props.planetId)
```

Comment:
so `props` is the next set of props, and `this.props` here is the last set, which will soon be set to equal `props`?